core/src/main/java/com/inrupt/client/core/DefaultClient.java

```diff
@@ -63,6 +64,7 @@ public final class DefaultClient implements Client {
 
     @Override
     public Client session(final Session session) {
+        Objects.requireNonNull(session, "Session may not be null!");
```

Comment:
I would be inclined to omit the check here and, instead, put it in the ctor:

https://github.com/inrupt/solid-client-java/blob/6acfe3a972bea515496d442d1f2395a758f6fe50/core/src/main/java/com/inrupt/client/core/DefaultClient.java#L60-L61